magicc_sim_plugins/src/step_camera.cpp

```diff
@@ -79,12 +85,22 @@ void StepCamera::OnUpdateParentSensor(){
 }
 
 void StepCamera::OnUpdate(const common::UpdateInfo&){
-   if(this->parentSensor->IsActive() && (this->world_->GetSimTime() - this->last_update_time_) >= (this->_updateRate) ){
-     // If we should have published a message, try and get a lock to wait for onUpdateParentSensor
-     if(!this->_updateLock.timed_lock(boost::posix_time::seconds(this->_updateRate))){
-         ROS_FATAL_STREAM("Update loop timed out waiting for the renderer.");
-     }
-   }
+    #if GAZEBO_MAJOR_VERSION >= 8
+        if(this->parentSensor->IsActive() && (this->world_->SimTime() - this->last_update_time_) >= (this->_updateRate) ){
+            // If we should have published a message, try and get a lock to wait for onUpdateParentSensor
+            if(!this->_updateLock.timed_lock(boost::posix_time::seconds(this->_updateRate))){
+                ROS_FATAL_STREAM("Update loop timed out waiting for the renderer.");
+            }
+        }
+    #else
+        if(this->parentSensor->IsActive() && (this->world_->GetSimTime() - this->last_update_time_) >= (this->_updateRate) ){
+        // If we should have published a message, try and get a lock to wait for onUpdateParentSensor
+            if(!this->_updateLock.timed_lock(boost::posix_time::seconds(this->_updateRate))){
+                ROS_FATAL_STREAM("Update loop timed out waiting for the renderer.");
+            }
+        }
+    #endif
+
```

Comment:
I can't see the difference between the two versions here.  Can you reduce the redundancy?